### ISSUE TITLE
add validation on _id field when upsert new doc

### ISSUE THREAD

**Issue by wdongyu** [elastic/elasticsearch#102981]
### Elasticsearch Version

any versions

### Java Version

java version "17.0.5" 2022-10-18 LTS

### OS Version

Darwin

### Problem Description

It seems that validate() in UpdateRequest does not include the limitation on _id, but IndexRequest does:

![image](https://github.com/elastic/elasticsearch/assets/23725216/66ab6e9c-5ab9-4ccb-8769-cab8e0a205d1)

In this scenario, when I add a new document through upsert, it will be executed successfully. But later when I reindex from the source index, it fail due to the limitation on _id, because reindex first search then add IndexRequests to bulk.

### Steps to Reproduce

1. index a doc with a long id, it fail:
`
curl --location --request POST 'localhost:9200/test_id/_bulk' \
--header 'Content-Type: application/json' \
--data-raw '{ "index" : {"_id" : "Ю10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8Ю10iS7hWAbPcH8πЮ10iS7hWAbPcH8ππЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ1" } }
{ "doc" : {"field2" : "value2"}}
'
`

2. upsert a doc with a long id, it success:
`
curl --location --request POST 'localhost:9200/test_id/_bulk' \
--header 'Content-Type: application/json' \
--data-raw '{ "update" : {"_id" : "Ю10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8Ю10iS7hWAbPcH8πЮ10iS7hWAbPcH8ππЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ10iS7hWAbPcH8πЮ1" } }
{ "doc" : {"field2" : "value2"}, "doc_as_upsert": true}
'
`

**Comment by wdongyu** [comment]
willing to open a pull request 😊

**Comment by benwtrent** [comment]
This is indeed surprising behavior to me! A PR here would be welcome.

I don't offhand know of anything that is abusing this unintended behavior :).

**Comment by Tejasker** [comment]
hey @benwtrent , @carlosdelest  I'm willing to contribute & raise PR 
at moment I am new to open source , a small gesture of help in further proceeding step is appreciated.
do let me know , how can I put my knowledge into real time implementation 

Thanks in Advance

**Comment by wdongyu** [comment]
@Tejasker I'm working on that, will raise a PR soon 😊

**Comment by Tejasker** [comment]
@wdongyu thanks for letting me 👍 !!
I thought of starting it  anyways ATB

**Comment by bohemia420** [comment]
@wdongyu , I have requested some changes. Also, you might have to adhere to Contributing Guidelines - chiefly:
```
Submit a pull request

Push your local changes to your forked copy of the repository and [submit a pull request](https://help.github.com/articles/using-pull-requests). In the pull request, choose a title which sums up the changes that you have made, and in the body provide more details about what your changes do. Also mention the number of the issue where discussion has taken place, eg "Closes #123".
```

just kindly add `Closes #102981`, otherwise your PR doesnt show up against this "Good first issue", and other contributors might duplicate the effort on this issue, rather than reviewing and closing your good work (=:

**Comment by wdongyu** [comment]
@bohemia420 LGTM, will update it
